docs/user-guide/expressions/lists.md

```diff
@@ -1,47 +1,102 @@
+`Polars` has first-class support for `List` columns: that is, columns where each row is a list of homogenous elements, of varying lengths. `Polars` also has an `Array` datatype, which is analogous to `numpy`'s `ndarray` objects, where the length is identical across rows.
 
-# Row wise computations
+Note: this is different from Python's `List` object, where the elements can be of any time. Polars can store these within columns, but as a generic `Object` datatype that doesn't have the special list manipulation features that we're about to discuss.
 
-This context is ideal for computing things in row orientation.
+# Powerful *List* manipulation
 
-Polars expressions work on columns that have the guarantee that they consist of homogeneous data.
-Columns have this guarantee, rows in a `DataFrame` not so much.
-Luckily we have a data type that has the guarantee that the rows are homogeneous: `pl.List` data type.
+Let's say we had the following data from different weather stations across a state. When the weather station is unable to get a result, an error code is recorded instead of the actual temperature at that time.
 
-Let's say we have the following data:
+{{code_block('user-guide/expressions/lists','weather_df',['DataFrame'])}}
+```python exec="on" result="text" session="user-guide/lists"
+--8<-- "python/user-guide/expressions/lists.py:setup"
+--8<-- "python/user-guide/expressions/lists.py:weather_df"
+```
+
+## Creating a *List* column
 
-{{code_block('user-guide/expressions/lists','dataframe',['DataFrame'])}}
+For the `weather` `DataFrame` created above, it's very likely we need to run some analysis on the temperatures that are captured by each station. To make this happen, we need to first be able to get individual temperature measurements. This is done by:
 
+{{code_block('user-guide/expressions/lists','string_to_list',['str.split'])}}
 ```python exec="on" result="text" session="user-guide/lists"
---8<-- "python/user-guide/expressions/lists.py:setup"
---8<-- "python/user-guide/expressions/lists.py:dataframe"
+--8<-- "python/user-guide/expressions/lists.py:string_to_list"
+```
+
+One way we could go post this would be to convert each temperature measurement into its own row:
+
+{{code_block('user-guide/expressions/lists','explode_to_atomic',['list.explode'])}}
+```python exec="on" result="text" session="user-guide/lists"
+--8<-- "python/user-guide/expressions/lists.py:explode_to_atomic"
+```
+
+However, in Polars, we often do not need to do this to operate on the `List` elements.
+
+## Operating on *List* columns
+
+Polars provides several standard operations on `List` columns. If we want the first three measurements, we can do a `head(3)`. The last three can be obtained via a `tail(3)`, or alternately, via `slice` (negative indexing is supported). We can also identify the number of observations via `lengths`. Let's see them in action:
+
+{{code_block('user-guide/expressions/lists','list_ops',['list_ops'])}}
```

Comment:
Use the third argument to highlight native functions you are using in the code snippet. You can add the function in API_REFERENCE_LINKS.yaml to create links to the documentation. If you only have python examples than only add the links for python

Reply:
Ah, this is why my first attempts to do so were not working. Thank you very much!